v6.5.0-rc.4 `pixi.js` package has invalid `@pixi/canvas-renderer` dependency

### Expected Behavior

After upgrading a project from the v6.4.2 to the v6.5.0-rc.4 I shouldn't have any errors.

### Current Behavior

```sh
Error: node_modules/@pixi/app/index.d.ts:4:37 - error TS2307: Cannot find module '@pixi/canvas-renderer' or its corresponding type declarations.

4 import type { CanvasRenderer } from '@pixi/canvas-renderer';
                                      ~~~~~~~~~~~~~~~~~~~~~~~
```

```sh
Error: node_modules/@pixi/settings/index.d.ts:3:37 - error TS2307: Cannot find module 'ismobilejs' or its corresponding type declarations.

3 import type { isMobileResult } from 'ismobilejs';
                                      ~~~~~~~~~~~~
```

`@pixi/app` has a dependency to `@pixi/canvas-renderer` that was not present in the `v6.4.2`, as `pixi.js` is described `PixiJS without the CanvasRenderer fallback, modern browsers only` I think this dependency should not be present.

As for the `isMobileResult`, I think it is related to #8502 that is not fully fixed yet.

### Steps to Reproduce

```sh
git clone git@github.com:Julien-Marcou/puzzle.git
cd puzzle/front
git checkout pixi-6.5
npm install
npm run start

```

### Environment

- **`pixi.js` version**: `v6.5.0-rc.4`

--- a/packages/app/src/ResizePlugin.ts
+++ b/packages/app/src/ResizePlugin.ts
@@ -1,7 +1,8 @@
-import type { CanvasRenderer } from '@pixi/canvas-renderer';
 import type { ExtensionMetadata, Renderer } from '@pixi/core';
 import { ExtensionType } from '@pixi/core';
 import type { IApplicationOptions } from './Application';
+
+type ResizeableRenderer = Pick<Renderer, 'resize'>;
 
 /**
  * Middleware for for Application's resize functionality
@@ -15,7 +16,7 @@ export class ResizePlugin
 
     public static resizeTo: Window | HTMLElement;
     public static resize: () => void;
-    public static renderer: Renderer | CanvasRenderer;
+    public static renderer: ResizeableRenderer;
     public static queueResize: () => void;
     private static _resizeId: number;
     private static _resizeTo: Window | HTMLElement;

--- a/packages/settings/src/utils/isMobile.ts
+++ b/packages/settings/src/utils/isMobile.ts
@@ -1,9 +1,39 @@
-// The ESM/CJS versions of ismobilejs only
-// exports the function for executing
-// designed for Node-only environments
 import isMobileCall from 'ismobilejs';
-import type { isMobileResult } from 'ismobilejs';
 
-const isMobile: isMobileResult = isMobileCall(globalThis.navigator);
+type isMobileResult = {
+    apple: {
+        phone: boolean;
+        ipod: boolean;
+        tablet: boolean;
+        universal: boolean;
+        device: boolean;
+    };
+    amazon: {
+        phone: boolean;
+        tablet: boolean;
+        device: boolean;
+    };
+    android: {
+        phone: boolean;
+        tablet: boolean;
+        device: boolean;
+    };
+    windows: {
+        phone: boolean;
+        tablet: boolean;
+        device: boolean;
+    };
+    other: {
+        blackberry: boolean;
+        blackberry10: boolean;
+        opera: boolean;
+        firefox: boolean;
+        chrome: boolean;
+        device: boolean;
+    };
+    phone: boolean;
+    tablet: boolean;
+    any: boolean;
+};
 
-export { isMobile };
+export const isMobile: isMobileResult = isMobileCall(globalThis.navigator);

